### DISCUSSION TITLE
Applying eigenstrain on a single grain

### DISCUSSION THREAD

**Discussion by shahul31** [neperfepx/FEPX#82] [Q&A] [answered]
Hello,

Could anyone help! Is it possible to apply an eigenstrain on a single crystal of a polycrystal cubic domain using FEPX?

Many thanks

**Comment by rquey** [comment, marked as answer]
No, it is not possible in the current version.

**Reply by shahul31** [reply]
Thank you for the answer. If you don't mind, could you please suggest some tool (FEM or FFT solver or something like that)

**Reply by rquey** [reply]
Ah, if you ask, I would suggest to implement it in FEPX...

**Reply by shahul31** [reply]
Ok. Thank you again for suggestion :)

**Comment by shahul31** [comment]
![Poly10_1112](https://user-images.githubusercontent.com/67755529/168591857-d5eff53d-9e2f-48e7-8202-1a7c95aec68d.gif)
Hello Romain Quey,

Could you please provide some insights on this, I am trying to develop a grain evolution model, for example, FCC to BCC like austenite to ferrite in which I can able to make grain growth as BCC surrounded by FCC grains with neper by increasing the grain weight of BCC at several steps, here FCC and BCC are different groups in neper, this (attached gif) is just grain growth without any actual physics involving. And so I would like to have a growing grain (only BCC grain) with applied strain in FEPX. So to implement this I can modify some pre-existing files or need to develop a new script? Your thoughts will be more helpful.
Thank you.

**Reply by mkasemer** [reply]
You could likely technically use the existing code to achieve _something_ similar to this, by applying boundary conditions directly to the nodes of the grain in question. However, your other grains are not going to disappear, they will just be compressed by the growth of the grain on which you place the nodal velocities.

Otherwise, I think what you really need to consider is some other code base. FEPX is made for deformation modeling, while this is more akin to recrystallization (phase field modeling might be more appropriate).

**Reply by shahul31** [reply]
Thank you for the clue, I am actually expecting the growing BCC to consume FCC and grows further - a phase transition to occur.

**Reply by mkasemer** [reply]
This is a much more difficult problem to consider than simply applying an eigenstrain. I suggest you need to consult literature to understand better what modeling capabilities currently exist to approach this problem, as well as the focus and limitations of crystal plasticity modeling.

**Reply by shahul31** [reply]
Ok I understand, thanks again. Have a good day. :)

**Reply by rquey** [reply]
Maybe a reference here: https://doi.org/10.1016/j.ijplas.2010.09.008
and another one: https://doi.org/10.1016/j.actamat.2020.06.032

**Reply by shahul31** [reply]
Thanks a lot for the references.
